bundle/phases/deploy.go

```diff
@@ -14,10 +17,90 @@ import (
 	"github.com/databricks/cli/bundle/permissions"
 	"github.com/databricks/cli/bundle/python"
 	"github.com/databricks/cli/bundle/scripts"
+	"github.com/databricks/cli/libs/cmdio"
+	terraformlib "github.com/databricks/cli/libs/terraform"
 )
 
+func approvalForDeploy(ctx context.Context, b *bundle.Bundle) (bool, error) {
+	if b.AutoApprove {
+		return true, nil
+	}
+
+	tf := b.Terraform
+	if tf == nil {
+		return false, fmt.Errorf("terraform not initialized")
+	}
+
+	// read plan file
+	plan, err := tf.ShowPlanFile(ctx, b.Plan.Path)
+	if err != nil {
+		return false, err
+	}
+
+	actions := make([]terraformlib.Action, 0)
+	for _, rc := range plan.ResourceChanges {
+		var actionType terraformlib.ActionType
+
+		switch {
+		case rc.Change.Actions.Delete():
+			actionType = terraformlib.ActionTypeDelete
+		case rc.Change.Actions.Replace():
+			actionType = terraformlib.ActionTypeRecreate
+		default:
+			// We don't need a prompt for non-destructive actions like creating
+			// or updating a schema.
+			continue
+		}
+
+		actions = append(actions, terraformlib.Action{
+			Action:       actionType,
+			ResourceType: rc.Type,
+			ResourceName: rc.Name,
+		})
+	}
+
+	// No restricted actions planned. No need for approval.
+	if len(actions) == 0 {
+		return true, nil
+	}
+
+	cmdio.LogString(ctx, "The following UC schemas will be deleted or recreated. Any underlying data may be lost:")
+	for _, action := range actions {
+		cmdio.Log(ctx, action)
+	}
+
+	if b.AutoApprove {
```

Comment:
Can be removed, you have the same at the top of the function

Reply:
Good catch! I removed the one at the top since I'd like to print the plan anyway for users to see in their CI logs.